modules/apps/dynamic-data-mapping/dynamic-data-mapping-service/src/main/java/com/liferay/dynamic/data/mapping/service/impl/DDMFieldLocalServiceImpl.java

```diff
@@ -487,8 +488,20 @@ public void updateDDMFormValues(
 			ddmFieldAttribute.setAttributeName(
 				ddmFieldAttributeInfo._attributeName);
 			ddmFieldAttribute.setLanguageId(ddmFieldAttributeInfo._languageId);
-			ddmFieldAttribute.setAttributeValue(
-				ddmFieldAttributeInfo._attributeValue);
+
+			String attributeValue = ddmFieldAttributeInfo._attributeValue;
+
+			DDMFormField ddmFormField = ddmFormFieldsMap.get(
+				ddmFieldAttributeInfo._ddmFieldInfo._fieldName);
+
+			if ((ddmFormField != null) &&
+				GetterUtil.getBoolean(
+					ddmFormField.getProperty("persistReadOnlyValue"))) {
```

Comment:
@carolmariaabb 

A question: The flag read-only of the DDMFormField doesn't reflect the evaluations from rules here, right?

Reply:
Actually, the question is: Why can't we use `readOnly` instead of this new propety `persistReadOnlyValue`

Reply:
@gabrielwas because this code `ddmFormFieldLocalService.updateDDMFormValues(...)` is also used by Web Content and Documents and Media.

Currently, a user can create a structure with a `readOnly` field, using the Data Engine Headless API, and send values ​​to this field when creating a Web Content, for example.

If I change to:

```
if ((ddmFormField != null) && ddmFormField.isReadOnly()) {
	ddmFieldAttribute.setAttributeValue(ddmFieldAttribute.getAttributeValue());
}
else{
	ddmFieldAttribute.setAttributeValue(ddmFieldAttributeInfo._attributeValue);
}
```

then I will not allow any `readOnly` field value to be updated. 

I really don't want to change the behavior of WC and D&M right now, even though I'm going to fix the behavior for readOnly fields. I think we need to talk to the Echo and Lima team before doing this.

Reply:
Good. Got it. Thx, @carolmariaabb !